pybossa/sched.py

```diff
@@ -218,15 +218,24 @@ def template_get_locked_task(project_id, user_id=None, user_ip=None,
             "Project {} - number of current users: {}"
             .format(project_id, user_count))
 
+        present_gold_task = not random.randint(0, 10)
         sql = query_factory(project_id, user_id, user_ip, external_uid,
-                            limit, offset, orderby, desc, rand_within_priority)
+                            limit, offset, orderby, desc, rand_within_priority,
+                            present_gold_task)
         rows = session.execute(sql, dict(project_id=project_id,
                                          user_id=user_id,
                                          limit=user_count + 5))
 
+        if not rows.rowcount and present_gold_task:
+            sql = query_factory(project_id, user_id, user_ip, external_uid,
+                                limit, offset, orderby, desc, rand_within_priority)
+            rows = session.execute(sql, dict(project_id=project_id,
+                                             user_id=user_id,
+                                             limit=user_count + 5))
+
         for task_id, taskcount, n_answers, timeout in rows:
             timeout = timeout or TIMEOUT
-            remaining = n_answers - taskcount
+            remaining = 9999 if present_gold_task else n_answers - taskcount
```

Comment:
You could also use `float('inf')` or `sys.maxint`.  I'm sure 9999 is beyond the scope of almost any application, but something like `float('inf')` stress the point being made here.

Reply:
agree

Reply:
you could also skip locking when the task is golden

Reply:
Or we can update the lock method to accept a kwarg e.g. `no_limit = True` so we don't have to make assumptions about the implementation.

Reply:
Or when the `remaining` is negative, that could mean no limit.